pennylane/measurements/sample.py

```diff
@@ -141,7 +141,7 @@ def process_samples(
 
         if self.obs is None:
             # if no observable was provided then return the raw samples
-            return samples if bin_size is None else samples.reshape(num_wires, bin_size, -1)
+            return samples if bin_size is None else samples.T.reshape(num_wires, bin_size, -1)
```

Comment:
I really don't like changing the `num_wires` axis here. @eddddddy would you like to check if it's possible to change this behaviour such that the `num_wires` axis is always the last axis?

It is not urgent, but it would simplify a LOT all the logic that use sampling results.

Reply:
The device has its own logic that swaps some axes, so the output of the QNode is the expected shape. The issue is that this logic impacts some other components of the device as well.

I don't really know why we decided to do things this way but changing it right now would be out-of-scope I think.